src/log/log.c

```diff
@@ -1970,6 +1970,10 @@ __log_write_internal(WT_SESSION_IMPL *session, WT_ITEM *record, WT_LSN *lsnp,
 	    myslot.slot != NULL)
 		ret = myslot.slot->slot_error;
 
+	if (LF_ISSET(WT_LOG_FLUSH))
+		WT_ASSERT(session, __wt_log_cmp(&log->write_lsn, &lsn) >= 0);
+	if (LF_ISSET(WT_LOG_FSYNC))
+		WT_ASSERT(session, __wt_log_cmp(&log->sync_lsn, &lsn) >= 0);
```

Comment:
@sueloverso I'm uncomfortable with ``if`` statements that only contain an assertion. I know it works with how we define ``WT_ASSERT`` in non-diagnostic builds, but it looks odd to me. Could these be refactored like:

``
WT_ASSERT(session, !LF_ISSET(WT_LOG_FLUSH) || __wt_log_cmp(&log->write_lsn, &lsn) >= 0);
``

To avoid the ``if`` statement? I know that makes the logic less clear (probably to the point where a comment is warranted). This is just a suggestion - if you prefer it as-is this is good to merge.